crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -601,6 +675,210 @@ struct MemoryPool {
     max_wasm_pages: u64,
 }
 
+/// A single slot handled by the memfd instance-heap mechanism.
+///
+/// The mmap scheme is:
+///
+/// base ==> (points here)
+/// - (image.offset bytes)   anonymous zero memory, pre-image
+/// - (image.len bytes)      CoW mapping of memfd heap image
+/// - (up to extension_offset)  anonymous zeor memory, post-image
+/// - (up to static_size)    heap expansion region; CoW mapping of per-slot memfd
+#[cfg(feature = "memfd-allocator")]
+#[derive(Debug)]
+pub struct MemFdSlot {
+    /// The base of the actual heap memory. Bytes at this address are
+    /// what is seen by the Wasm guest code.
+    base: usize,
+    /// The maximum static memory size, plus post-guard.
+    static_size: usize,
```

Comment:
While I don't think this is really an issue this feels morally to me like the size here shouldn't include the guard size. I naively expect that the `MemFdSlot` is only responsible for the linear memory, and the guard pages are concerns of other layers of the stack.

Reply:
Yeah, I agree in principle; this is just following how the rest of the pooling allocator works. Could definitely be cleaned up in a followup.